src/Spectre.Angular2Client/src/app/preparation/preparation.component.ts

```diff
@@ -30,14 +31,43 @@ import { PreparationRoutingModule } from './preparation-routing.module';
 export class PreparationComponent implements OnInit {
 
   id: number = null;
+  public Layout: any;
+  public Data: any;
+  public Options: any;
 
   constructor(
       private route: ActivatedRoute,
+      private spectrumService: SpectrumService
   ) { }
 
   ngOnInit() {
       this.route.params.subscribe(params => {
-          this.id = Number.parseInt(params['id']);
+        console.log('[PreparationComponent] ngOnInit');
+        this.id = Number.parseInt(params['id']);
+        console.log('[PreparationComponent] parsed id');
+        this.spectrumService.get(this.id, 1).subscribe(((spectrum) => {
```

Comment:
Może to co jest w subscribe dać do osobnej funkcji? Ale można to zrobić wtedy, kiedy będzie się robiło to co jest w issues